nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/LogString.java

```diff
@@ -0,0 +1,208 @@
+    private List<PropertyDescriptor> supportedDescriptors;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> procRels = new HashSet<>();
+        procRels.add(REL_SUCCESS);
+        relationships = Collections.unmodifiableSet(procRels);
+
+        // descriptors
+        final List<PropertyDescriptor> supDescriptors = new ArrayList<>();
+        supDescriptors.add(LOG_LEVEL);
+        supDescriptors.add(LOG_PREFIX);
+        supDescriptors.add(LOG_MESSAGE);
+        supportedDescriptors = Collections.unmodifiableList(supDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return supportedDescriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+
+        final String logLevelValue = context.getProperty(LOG_LEVEL).getValue().toLowerCase();
+
+        final MessageLogLevel logLevel;
+        try {
+            logLevel = MessageLogLevel.valueOf(logLevelValue);
+        } catch (Exception e) {
+            throw new ProcessException(e);
+        }
+
+        final ComponentLog logger = getLogger();
+        boolean isLogLevelEnabled = false;
+        switch (logLevel) {
+            case trace:
+                isLogLevelEnabled = logger.isTraceEnabled();
+                break;
+            case debug:
+                isLogLevelEnabled = logger.isDebugEnabled();
+                break;
+            case info:
+                isLogLevelEnabled = logger.isInfoEnabled();
+                break;
+            case warn:
+                isLogLevelEnabled = logger.isWarnEnabled();
+                break;
+            case error:
+                isLogLevelEnabled = logger.isErrorEnabled();
+                break;
+        }
+
+        if (!isLogLevelEnabled) {
+            transferChunk(session);
+            return;
+        }
+
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        processFlowFile(logger, logLevel, flowFile, context);
+        session.transfer(flowFile, REL_SUCCESS);
+    }
+
+    private void processFlowFile(
+            final ComponentLog logger,
+            final MessageLogLevel logLevel,
+            final FlowFile flowFile,
+            final ProcessContext context) {
+
+        String logPrefix = context.getProperty(LOG_PREFIX).evaluateAttributeExpressions(flowFile).getValue();
+        String logMessage = context.getProperty(LOG_MESSAGE).evaluateAttributeExpressions(flowFile).getValue();
+
+        String messageToWrite;
+        if (StringUtil.isBlank(logPrefix)) {
+            messageToWrite = logMessage;
+        } else {
+            messageToWrite = String.format("%s %s", logPrefix, logMessage);
```

Comment:
Do you think the extra space between the prefix and message could throw off the user at first? Certainly if they look in the logs they could see there is a space between what they have specified for the prefix vs the message, but I'm wondering if it is best to omit the space or update the property documentation to make it clear there will or will not be whitespace between the prefix and message. I mention it only because I left a space at the end of my prefix and tried to search for the prefix+message and didn't find it right away. Again, not a requirement, just a suggestion/question :)